src/main/kotlin/org/sd_jwt/SdJwt.kt

```diff
@@ -166,13 +161,12 @@ fun createSdClaims(
  */
 inline fun <reified T> createCredential(
     userClaims: T,
-    issuerKey: JWK,
+    signer: SdJwtSigner,
     holderPubKey: JWK? = null,
     discloseStructure: T? = null,
     sdJwtHeader: SdJwtHeader = SdJwtHeader(),
```

Comment:
I guess we don't need the sdJwtHeader anymore right?

Reply:
Yes, that is possible, but it also required a change at the `SdJwtSigner`; it now can be provided with a `SdJwtHeader` and will then return the complete `JWSHeader` including the potential typ and cty header parameters using the `sdJwtHeader()` method. Let me know what you think of this.

Reply:
I think it's better now but would it be possible to just pass a ``JWSHeader`` object to the signer? That would give the user complete flexibility over the header. Or do you think that does not make sense?

Reply:
The ``JWSHeader`` object should be an optional argument.